package/MDAnalysis/analysis/hole.py

```diff
@@ -356,6 +362,7 @@ def plot(self, **kwargs):
         ax.set_ylabel(r"HOLE radius $r$ ($\AA$)")
         if kw['label']:
             ax.legend(loc="best")
+        return ax
```

Comment:
API for receiving an axis object should be.

```python
# have the extra line here to later change to explicit API parameter.
ax = kwargs.pop('ax', None)
if ax is None:
    ax = plt.gca()
```

This will plot into the last used axis object. If none was used it creates a new one like `plt.subplots(111)`. This is the best practice encouraged by matplotlib and how other packages behave.